lib/chipmunk/package_storage.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Chipmunk
+  # PackageStorage is a factory for proxies to a Package's underlying storage.
+  # An instance supports a named set of format types, mapping each to a storage
+  # proxy class and creating an appropriate proxy instance for a package.
+  class PackageStorage
+
+    # Create a PackageStorage instance.
+    #
+    # @param volumes [Array<Chipmunk::Volume>] the storage volumes where packages are stored
+    def initialize(volumes:)
+      @volumes = Hash[volumes.map {|volume| [volume.name, volume] }]
+    end
+
+    # Retrieve the storage proxy for a given package. The package must be stored and of a registered format.
+    # @param package [Package] the package for which to retrieve a storage proxy
+    # @raise [Chipmunk::PackageNotStoredError] if the package is not stored
+    # @raise [Chipmunk::UnsupportedFormatError] if the package format is not registered
+    def for(package)
+      raise Chipmunk::PackageNotStoredError, package unless package.stored?
```

Comment:
Not sure we need this; `volume_for` should already raise something if the package is not in fact stored, and eventually we probably want to rethink things so `stored?` can go away.

Reply:
In playing with this after discussion, I've added a raise to volume#get, but that doesn't solve this case. We either have to have this or a nil-check in volume_for. I'd rather leave this here for now and drop it once we distinguish between deposits and preserved packages (giving a non-nil storage_volume invariant).